### ISSUE TITLE
[REQUEST] Add support an external torrent clients (Transmission Daemon, qBittorrent and etc.)

### ISSUE THREAD

**Issue by Maks1mS** [hydralauncher/hydra#667]
### Is your feature request related to a problem? Please describe.

In my daily life, I use the Transmission Daemon on my NAS and would like to be able to use external torrent clients as well. One of the main benefits for me would be the ability to download games even when my PC is turned off.

### Describe the solution you'd like

I would like to be able to use external torrent clients for downloading. It would be nice to start with Transmission. I also know that qBittorrent has an API.

### Additional context

**Transmission** 

Transmission's RPC specification: https://github.com/transmission/transmission/blob/main/docs/rpc-spec.md
One of the libraries that implements Transmission's RPC: https://www.npmjs.com/package/transmission-promise

**qBittorrent**

qBittorrent WebUI API:  https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-(qBittorrent-4.1)

**Comment by bpavuk** [comment]
you can install the game as usual and then add it to the Hydra library

**Comment by JackEnx** [comment]
It seems like something very interesting, can you download files via torrent directly to the NAS? I don't think it's possible at the moment because it involves a certain level of research and it seems like something that wouldn't be used soo much (implemented by the core team, nothing prevents someone implementing it), perhaps in the future

**Comment by Maks1mS** [comment]
> It seems like something very interesting, can you download files via torrent directly to the NAS?

Yes, I do download files directly to my NAS and then access them using SMB. In my case, the folder is mounted as a regular file system, so it looks like that it functions similarly to an external client on my local machine with a local file system. However, this is just one option. Other users may prefer to access the files using SMB or HTTP, for example.

> it seems like something that wouldn't be used soo much

Of course, I suspected it. But without creating this issue, we really don't know it. :)

> (implemented by the core team, nothing prevents someone implementing it)

I tried to implement this, but I abandoned the idea for the following reasons:
1. There are two download options in one class. I was afraid to break it and there was no way to check the Real Debrid
2. Started migration from aria2c to libtorrent with code refactoring.
3. In addition to implementation, this requires support. It may be discontinued in the future due to low usage. Also, if there was a plugin system, maybe it would be a good plugin that core team would not be responsible for.

Maybe in the future, I will return to this.

**Comment by Masong19hippows** [comment]
I want to bump this because its a genuine feature that really limits hydra. I think hydra could take inspiration from established arr services like [Radarr](https://github.com/Radarr/Radarr) and [Sonarr](https://github.com/Sonarr/Sonarr) to further the community base for this. With torrenting, there are alot of options and configuration choices that are best left to the user, but Hydra's built in libtorrent implementation lacks these configuration options. I think that in order to gain support for these configuration options, it would be much more simple to leave it up to the end-users already configured setup rather than take the time to implement it by hand. This approach also has the benefit of being expandable to other torrenting alternatives if structured properly. Usenet is also a very popular download option and services that have this structure are finding it very easy to incorporate Usenet into their service.

A good example of this is proxy support. You can see from libtorrent/qbittorrent (same general authors) that proxy support is a nightmare to setup because of so many edgecases. But, its a necessary requirement for some people in order to get general internet access. This is especially the case in Asian countries where the general internet is restricted and people need proxies to bypass those restrictions. 

I think the success of other services like radarr and sonar prove how succesfull this approach to torrenting can be. As it stands, Hydra is trying to be a replacement for solutions like qbittorrent and transmission, but it just lacks alot of the general configuration options to allow for this.

**Comment by Wamy-Dev** [comment]
I personally like the idea of this issue. I enjoy Sonarr and Radarr and something similar for games would be awesome. I like the idea of Usenet, although not very popular, has many positive benefits compared to torrents. 

_Disclaimer: I own torbox.app which has a Usenet feature (which would be awesome if integrated into Hydra)_

Even though I am the owner of a torrent downloading service, allowing users more customizability to how torrents are downloaded is something Hydra is lacking. 

Something like an external client would likely still need to reside on the machine that has Hydra or have some sort adapter that sends the completed file to the computer with Hydra. This is likely the most difficult and time consuming part of this issue.
